### PR TITLE
Add option to limit number of frames captured

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Use `DD_EXCEPTION_REPLAY_MAX_FRAMES_TO_CAPTURE` to limit number of frame in user code (i.e. not third-party code) to instrument and capture per exception. Default value is 3

# Motivation
limit the amount of snapshots generated

# Additional Notes

Jira ticket: [DEBUG-2068]

[DEBUG-2068]: https://datadoghq.atlassian.net/browse/DEBUG-2068?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java [hunk 1]
```diff
@@ -88,4 +87,37 @@ void lastCapture() {
...
+
+  RuntimeException level1() {
+    return level2();
+  }
+
+  RuntimeException level2() {
+    return level3();
+  }
+
+  RuntimeException level3() {
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java:120]: should we have 4 levels to see that default 3 works?

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java [hunk 2]
```diff
@@ -88,4 +87,37 @@ void lastCapture() {
...
+                    "sun.",
+                    "com.sun.",
+                    "org.gradle.",
+                    "worker.org.gradle.",
+                    "org.junit.")
+                .collect(Collectors.toSet()));
+    ExceptionProbeManager exceptionProbeManager = new ExceptionProbeManager(classNameFiltering);
+    exceptionProbeManager.createProbesForException(deepException.getStackTrace());
+    assertEquals(
+        Config.get().getDebuggerExceptionMaxCapturedFrames(),
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java:108]: this test breaks if we change the default value - maybe better to hard code this to 3 for the test?

**Review by shatzi — APPROVED** [review]
very cool implementation

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java:120]: we have 4 user level with test in itself :)

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeManagerTest.java]: done

**Pull request merged**